pkg/reader/config_poller.go

```diff
@@ -65,11 +67,333 @@ func newConfigPoller(
 	refreshPeriod time.Duration,
 ) *configPoller {
 	return &configPoller{
-		chainCaches:   make(map[cciptypes.ChainSelector]*chainCache),
-		refreshPeriod: refreshPeriod,
-		reader:        reader,
-		lggr:          lggr,
+		chainCaches:       make(map[cciptypes.ChainSelector]*chainCache),
+		refreshPeriod:     refreshPeriod,
+		reader:            reader,
+		lggr:              lggr,
+		knownSourceChains: make(map[cciptypes.ChainSelector]map[cciptypes.ChainSelector]bool),
+		stopChan:          make(chan struct{}),
+	}
+}
+
+// startBackgroundPolling starts a goroutine that periodically refreshes the cache
+func (c *configPoller) startBackgroundPolling() {
+	c.wg.Add(1)
+	go func() {
+		defer c.wg.Done()
+		ticker := time.NewTicker(c.refreshPeriod)
+		defer ticker.Stop()
+
+		for {
+			select {
+			case <-ticker.C:
+				c.refreshAllKnownChains()
+			case <-c.stopChan:
+				return
+			}
+		}
+	}()
+}
+
+// Add Start method to configPoller
+func (c *configPoller) Start() {
+	c.startBackgroundPolling()
+
+	c.lggr.Info("Background poller started")
```

Comment:
Some field I would add to this log (unless they're already part of logger fields): `c.refreshPeriod`.